yt_dlp/extractor/gamejolt.py

```diff
@@ -0,0 +1,544 @@
+            'uploader_id': user_data.get('username'),
+            'uploader_url': 'https://gamejolt.com' + user_data['url'] if user_data.get('url') else None,
+            'categories': [try_get(category, lambda x: '%s - %s' % (x['community']['name'], x['channel'].get('display_title') or x['channel']['title']))
+                           for category in post_data.get('communities' or [])],
+            'tags': traverse_obj(
+                lead_content, ('content', ..., 'content', ..., 'marks', ..., 'attrs', 'tag'), expected_type=str_or_none),
+            'like_count': int_or_none(post_data.get('like_count')),
+            'comment_count': int_or_none(post_data.get('comment_count'), default=0),
+            'timestamp': int_or_none(post_data.get('added_on'), scale=1000),
+            'release_timestamp': int_or_none(post_data.get('published_on'), scale=1000),
+            '__post_extractor': self.extract_comments(post_data.get('id'), post_id)
+        }
+
+        # TODO: Handle multiple videos/embeds?
+        video_data = traverse_obj(post_data, ('videos', ...), expected_type=dict, get_all=False) or {}
+        formats, subtitles, thumbnails = [], {}, []
+        for media in video_data.get('media') or []:
+            media_url, mimetype, ext, media_id = media['img_url'], media.get('filetype', ''), determine_ext(media['img_url']), media.get('type')
+            if mimetype == 'application/vnd.apple.mpegurl' or ext == 'm3u8':
+                hls_formats, hls_subs = self._extract_m3u8_formats_and_subtitles(media_url, post_id, 'mp4', m3u8_id=media_id)
+                formats.extend(hls_formats)
+                subtitles.update(hls_subs)
+            elif mimetype == 'application/dash+xml' or ext == 'mpd':
+                dash_formats, dash_subs = self._extract_mpd_formats_and_subtitles(media_url, post_id, mpd_id=media_id)
+                formats.extend(dash_formats)
+                subtitles.update(dash_subs)
+            elif 'image' in mimetype:
+                thumbnails.append({
+                    'id': media_id,
+                    'url': media_url,
+                    'width': media.get('width'),
+                    'height': media.get('height'),
+                    'filesize': media.get('filesize'),
+                })
+            else:
+                formats.append({
+                    'format_id': media_id,
+                    'url': media_url,
+                    'width': media.get('width'),
+                    'height': media.get('height'),
+                    'filesize': media.get('filesize'),
+                    'acodec': 'none' if 'video-card' in media_url else None,
+                })
+
+        if formats:
+            return {
+                **info_dict,
+                'formats': formats,
+                'subtitles': subtitles,
+                'thumbnails': thumbnails,
+                'view_count': int_or_none(video_data.get('view_count')),
+            }
+
+        gif_entries = []
```

Comment:
Forgot to check this first, but GIF extraction breaks with an `extractor_key` KeyError if nested from a post list (easiest way to check is with `https://gamejolt.com/c/fnf --playlist-end 10`)

Reply:
Relevant debug output:
```
>yt-dlp https://gamejolt.com/c/fnf --playlist-end 10 -v --skip-download
[debug] Command-line config: ['https://gamejolt.com/c/fnf', '--playlist-end', '10', '-v', '--skip-download']
[debug] Encodings: locale cp1252, fs utf-8, out utf-8, err utf-8, pref cp1252
[debug] yt-dlp version 2021.12.01 [91f071af6] (source)
[debug] Lazy loading extractors is disabled
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Python version 3.9.5 (CPython 64bit) - Windows-10-10.0.19043-SP0
[debug] exe versions: ffmpeg N-104865-g571e8ca2dd-20211215 (setts), phantomjs 2.1.1
[debug] Optional libraries: Cryptodome, keyring, mutagen, sqlite, websockets
[debug] Proxy map: {}
[debug] [GameJoltCommunity] Extracting URL: https://gamejolt.com/c/fnf
[GameJoltCommunity] fnf: Downloading community info
[GameJoltCommunity] fnf: Downloading channel info
[download] Downloading playlist: Friday Night Funkin' - featured
[GameJoltCommunity] fnf: Downloading community posts
[GameJoltCommunity] aqkwv3ma: Downloading m3u8 information
[GameJoltCommunity] rfkr6vfj: Downloading m3u8 information
[GameJoltCommunity] ystbtagu: Downloading article metadata
[GameJoltCommunity] rbdzdwfq: Downloading m3u8 information
[GameJoltCommunity] rbdzdwfq: Downloading m3u8 information
[GameJoltCommunity] rbdzdwfq: Downloading m3u8 information
[GameJoltCommunity] fnf: Downloading community posts page 2
[GameJoltCommunity] playlist Friday Night Funkin' - featured: Downloading 10 videos
...
[download] Downloading video 10 of 10
[download] Downloading playlist: Hello everyone i'm developing a pixel art style mod for FNF, and i'd like you to give me your opinion about the demo I made, this so that the beta is much better and with more quality.
 Link demo:

https://www.mediafire.com/file/74ihp5s9ikq42vj/FNF%2527_Vs_Unknown_UPDATE_DEMO.rar/file

I hope you like! :]
[GameJolt] playlist Hello everyone i'm developing a pixel art style mod for FNF, and i'd like you to give me your opinion about the demo I made, this so that the beta is much better and with more quality.
 Link demo:

https://www.mediafire.com/file/74ihp5s9ikq42vj/FNF%2527_Vs_Unknown_UPDATE_DEMO.rar/file

I hope you like! :]: Collected 1 videos; downloading 1 of them
[download] Downloading video 1 of 1
ERROR: 'extractor_key'
Traceback (most recent call last):
  File "\yt_dlp\YoutubeDL.py", line 1337, in wrapper
    return func(self, *args, **kwargs)
  File "\yt_dlp\YoutubeDL.py", line 1744, in __process_iterable_entry
    return self.process_ie_result(
  File "\yt_dlp\YoutubeDL.py", line 1540, in process_ie_result
    return self.__process_playlist(ie_result, download)
  File "\yt_dlp\YoutubeDL.py", line 1718, in __process_playlist
    'extractor_key': ie_result['extractor_key'],
KeyError: 'extractor_key'

[download] Finished downloading playlist: Friday Night Funkin' - featured
```